pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -99,49 +104,34 @@ func (c *Cluster) prepareStorageClassDeviceSets(config *provisionConfig) []rookv
 
-	// return old labeled pvc, if we find any
-	if len(oldPresentPVCs.Items) == 1 {
-		logger.Debugf("old labeled pvc %q found", oldPresentPVCs.Items[0].Name)
-		c.updatePVCIfChanged(pvc, &oldPresentPVCs.Items[0])
-		return &oldPresentPVCs.Items[0], nil
+	// check for the existence of the pvc
+	existingPVC, ok := existingPVCs[pvcStorageClassDeviceSetPVCId]
+	if !ok {
+		// The old name of the PVC didn't exist, now try the new PVC name and label
+		pvcStorageClassDeviceSetPVCId = deviceSetPVCID(storageClassDeviceSetName, pvcTemplate.GetName(), setIndex)
+		existingPVC = existingPVCs[pvcStorageClassDeviceSetPVCId]
 	}
+	pvc := makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, setIndex, pvcTemplate)
 
-	// check again with the new label for the presence of updated pvc
-	pvcStorageClassDeviceSetPVCId, pvcStorageClassDeviceSetPVCIdLabelSelector = makeStorageClassDeviceSetPVCIDNew(storageClassDeviceSetName, pvcTemplate.GetName(), setIndex)
-	pvc = makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId, setIndex, pvcTemplate)
-	presentPVCs, err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.clusterInfo.Namespace).List(metav1.ListOptions{LabelSelector: pvcStorageClassDeviceSetPVCIdLabelSelector})
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to list pvc %s for storageClassDeviceSet %s", pvcStorageClassDeviceSetPVCIdLabelSelector, storageClassDeviceSetName)
+	if existingPVC != nil {
+		logger.Infof("OSD PVC %q already exists", existingPVC.Name)
+
+		// Update the PVC in case the size changed
+		c.updatePVCIfChanged(pvc, existingPVC)
+		return existingPVC, nil
 	}
 
-	presentPVCsNum := len(presentPVCs.Items)
 	// No PVC found, creating a new one
-	if presentPVCsNum == 0 {
-		deployedPVC, err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.clusterInfo.Namespace).Create(pvc)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create pvc %q for storageClassDeviceSet %q", pvc.GetGenerateName(), storageClassDeviceSetName)
-		}
-		logger.Debugf("just created pvc %q", deployedPVC.Name)
-		return deployedPVC, nil
-		// The PVC is already present
-	} else if presentPVCsNum == 1 {
-		logger.Debugf("already present pvc %q", presentPVCs.Items[0].Name)
-		c.updatePVCIfChanged(pvc, &presentPVCs.Items[0])
-
-		// Updating with the new label
-		return &presentPVCs.Items[0], nil
+	deployedPVC, err := c.context.Clientset.CoreV1().PersistentVolumeClaims(c.clusterInfo.Namespace).Create(pvc)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create pvc %q for storageClassDeviceSet %q", pvc.GetGenerateName(), storageClassDeviceSetName)
 	}
-	// More than one PVC exists with same labelSelector
-	return nil, errors.Errorf("more than one PVCs exists with label %q, pvcs %q", pvcStorageClassDeviceSetPVCIdLabelSelector, presentPVCs)
+	logger.Infof("successfully provisioned PVC %q", deployedPVC.Name)
```

Comment:
I liked the previous info:

```
logger.Infof("successfully provisioned pvc %q for VolumeClaimTemplates %q for storageClassDeviceSet %q of set %v", pvc.GetName(), pvcTemplate.GetName(), storageClassDeviceSet.Name, i)
```

What do you think of adding it back here?

Reply:
The PVC name already contains all of that info so it was actually redundant.